libbeat/outputs/transport/tls.go

```diff
@@ -2,44 +2,262 @@ package transport
 
 import (
 	"crypto/tls"
+	"crypto/x509"
+	"errors"
 	"fmt"
 	"net"
+	"sync"
 	"time"
 )
 
-func TLSDialer(tlscfg *tls.Config, timeout time.Duration, forward Dialer) Dialer {
-	if tlscfg == nil {
-		return forward
+type TLSConfig struct {
+	Versions []TLSVersion
```

Comment:
wanted to added comments after adding all required fields. Thanks for pointing out.